packages/ui/src/accounts/hooks/useHasRequiredStake.ts

```diff
@@ -0,0 +1,67 @@
+import { useEffect, useState, useMemo } from 'react'
+
+import { useAccounts } from '@/accounts/hooks/useAccounts'
+import { useBalance } from '@/accounts/hooks/useBalance'
+import { useBalances } from '@/accounts/hooks/useBalances'
+import { useTotalBalances } from '@/accounts/hooks/useTotalBalances'
+import { Account } from '@/accounts/types'
+import { Address } from '@/common/types'
+
+export const useHasRequiredStake = (address: Address, stake: number) => {
```

Comment:
This check will be performed before the user selects an account for staking, so we can't assume one account only.